From 93e97d3514f19126780a3310662dcd4da77c4789 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 23 Apr 2024 17:39:57 +0200
Subject: [PATCH 0809/1695] dt-bindings: stm32_tamp: add STM32MP21 tamper
 bindings

Add STM32MP21 specific tamper bindings.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c00a612c268a137234f7ac26d616c8a3a0fcba17 dt-bindings: stm32_tamp: add STM32MP21 tamper bindings]
Change-Id: Ib13511277fd9d6ef34763f35040ca3c02509c9de
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/376034
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 .../dt-bindings/tamper/st,stm32mp21-tamp.h    | 37 +++++++++++++++++++
 1 file changed, 37 insertions(+)
 create mode 100644 core/include/dt-bindings/tamper/st,stm32mp21-tamp.h

diff --git a/core/include/dt-bindings/tamper/st,stm32mp21-tamp.h b/core/include/dt-bindings/tamper/st,stm32mp21-tamp.h
new file mode 100644
index 000000000..32f07be12
--- /dev/null
+++ b/core/include/dt-bindings/tamper/st,stm32mp21-tamp.h
@@ -0,0 +1,37 @@
+/* SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-2-Clause) */
+/*
+ * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+ * Author: Gatien Chevallier <gatien.chevallier@foss.st.com>
+ */
+
+/* Internal Tampers */
+#define INT_TAMPER_BKUP_DOMAIN_THRESHOLD	1
+#define INT_TAMPER_TEMPERATURE_MONITORING	2
+#define INT_TAMPER_LSE_MONITORING		3
+#define INT_TAMPER_HSE_MONITORING		4
+#define INT_TAMPER_RTC_CALENDAR_OVERFLOW	5
+#define INT_TAMPER_JTAG_SWD_ACCESS		6
+#define INT_TAMPER_ADC2_ANALOG_WD_1		7
+#define INT_TAMPER_MONOTONIC_COUNTER_1		8
+#define INT_TAMPER_CRYPTO_PERIPH_FAULT		9
+#define INT_TAMPER_MONOTONIC_COUNTER_2		10
+#define INT_TAMPER_IWDG3_RESET			11
+#define INT_TAMPER_ADC2_ANALOG_WD_2		12
+#define INT_TAMPER_ADC2_ANALOG_WD_3		13
+#define INT_TAMPER_RIFSC_BSEC_DBG_FAULT		14
+#define INT_TAMPER_IWDG1_RESET			15
+#define INT_TAMPER_BOOTROM_FAULT		16
+
+/* External Tampers */
+#define EXT_TAMPER_1				1
+#define EXT_TAMPER_2				2
+#define EXT_TAMPER_3				3
+#define EXT_TAMPER_4				4
+#define EXT_TAMPER_5				5
+#define EXT_TAMPER_6				6
+#define EXT_TAMPER_7				7
+#define EXT_TAMPER_8				8
+
+/* Tamper mode */
+#define TAMPER_CONFIRMED_MODE			1
+#define TAMPER_POTENTIAL_MODE			2
-- 
2.47.2

